SharedSources/MediaLibraryModel/MediaModel.swift

```diff
@@ -35,12 +35,34 @@ extension MediaModel {
 
 // MARK: - Helpers
 
+extension MediaModel {
+    /// Swap the given [VLCMLMedia] to the cached array.
+    /// This only swaps media with the same VLCMLIdentifiers
+    /// - Parameter medias: To be swapped medias
+    /// - Returns: New array of `VLCMLMedia` if changes have been made, else return a unchanged cached version.
+    func swapMedias(with medias: [VLCMLMedia]) -> [VLCMLMedia] {
+        var newFiles = files
+
+        // FIXME: This should be handled in a thread safe way
+        for media in medias {
+            for (currentMediaIndex, file) in files.enumerated()
+                where file.identifier() == media.identifier() {
+                    newFiles.remove(at: currentMediaIndex)
+                    newFiles.insert(media, at: currentMediaIndex)
```

Comment:
Opinion welcome if someone has a cleaver idea on how to swap these.
The only idea I have either this or: 

`swap(&newFiles[currentMediaIndex], &media)`

However, if we want the std lib version, I would need to set media as mutable.